test/integration/targets/docker_host_facts/tasks/test_host_facts.yml

```diff
@@ -0,0 +1,278 @@
+
+  - name: Get info on Docker host and get disk usage with verbose output
+    docker_host_facts:
+      disk_usage: yes
+      verbose_output: yes
+    register: output
+
+  - name: assert reading docker host facts when docker is running and get disk usage with verbose output
+    assert:
+      that:
+         - 'output.docker_host_facts.Name is string'
+         - 'output.docker_containers_list is not defined'
+         - 'output.docker_networks_list is not defined'
+         - 'output.docker_volumes_list is not defined'
+         - 'output.docker_images_list is not defined'
+         - 'output.docker_disk_usage.LayersSize is number'
+         - 'output.docker_disk_usage.BuilderSize is number'
+
+  - name: Get info on Docker host, disk usage and get all lists together
+    docker_host_facts:
+      volumes: yes
+      containers: yes
+      networks: yes
+      images: yes
+      disk_usage: yes
+    register: output
+
+  - name: assert reading docker host facts when docker is running, disk usage and get lists together
+    assert:
+      that:
+         - 'output.docker_host_facts.Name is string'
+         - 'output.docker_containers_list[0].Image is string'
+         - 'output.docker_containers_list[0].ImageID is not defined'
+         - 'output.docker_networks_list[0].Id is string'
+         - 'output.docker_networks_list[0].Created is not defined'
+         - 'output.docker_volumes_list[0].Name is string'
+         - 'output.docker_volumes_list[0].Mountpoint is not defined'
+         - 'output.docker_images_list[0].Id is string'
+         - 'output.docker_images_list[0].ParentId is not defined'
+         - 'output.docker_disk_usage.LayersSize is number'
+         - 'output.docker_disk_usage.BuilderSize is not defined'
+
+  - name: Get info on Docker host, disk usage and get all lists together with verbose output
+    docker_host_facts:
+      volumes: yes
+      containers: yes
+      networks: yes
+      images: yes
+      disk_usage: yes
+      verbose_output: yes
+    register: output
+
+  - name: assert reading docker host facts when docker is running and get disk usage with verbose output
+    assert:
+      that:
+         - 'output.docker_host_facts.Name is string'
+         - 'output.docker_containers_list[0].Image is string'
+         - 'output.docker_containers_list[0].ImageID is string'
+         - 'output.docker_networks_list[0].Id is string'
+         - 'output.docker_networks_list[0].Created is string'
+         - 'output.docker_volumes_list[0].Name is string'
+         - 'output.docker_volumes_list[0].Mountpoint is string'
+         - 'output.docker_images_list[0].Id is string'
+         - 'output.docker_images_list[0].ParentId is string'
+         - 'output.docker_disk_usage.LayersSize is number'
+         - 'output.docker_disk_usage.BuilderSize is number'
+
+  always:
+  - name: Delete container
+    docker_container:
+      image: alpine:3.8
+      command: '/bin/sh -c "sleep 10m"'
```

Comment:
For deletion, you need neither `image` nor `command`, just `name` and `state: absent`. You can also add `force_kill: yes` (that will simply kill the container without trying to shut it down for 10 seconds or so.)

Reply:
It's straight from `docker_container` integration tests playbook, same as the prefix for names. But it, of course, can be simplified :)

Reply:
The cleanup code of the docker_container tests doesn't have it, it's only used in test-specific "local" cleanup:
https://github.com/ansible/ansible/blob/devel/test/integration/targets/docker_container/tasks/main.yml#L20-L23